homeassistant/components/binary_sensor/rest.py

```diff
@@ -77,9 +77,17 @@ def is_on(self):
             return False
 
         if self._value_template is not None:
-            self.rest.data = template.render_with_possible_json_value(
+            self._state = template.render_with_possible_json_value(
```

Comment:
Why does this set the state and later you return data? The is_on method should not assign state but just return it. This logic should happen in the update method